mslib/mswms/wms.py

```diff
@@ -211,6 +214,114 @@ def __init__(self):
+            if all_plots:
+                # Imports here due to some circular import issue if imported too soon
+                from mslib.mswms import mpl_hsec_styles, mpl_vsec_styles, mpl_lsec_styles
+
+                dataset = [next(iter(mss_wms_settings.data))]
+                mss_wms_settings.register_horizontal_layers = [
+                    (plot[1], dataset) for plot in inspect.getmembers(mpl_hsec_styles, inspect.isclass)
+                    if not any(x in plot[0] or x in str(plot[1]) for x in ["Abstract", "Target", "fnord"])
+                ]
+                mss_wms_settings.register_vertical_layers = [
+                    (plot[1], dataset) for plot in inspect.getmembers(mpl_vsec_styles, inspect.isclass)
+                    if not any(x in plot[0] or x in str(plot[1]) for x in ["Abstract", "Target", "fnord"])
+                ]
+                mss_wms_settings.register_linear_layers = [
+                    (plot[1], dataset) for plot in inspect.getmembers(mpl_lsec_styles, inspect.isclass)
+                ]
+                self.__init__()
+
+            location = DOCS_LOCATION if sphinx else STATIC_LOCATION
+            if clear and os.path.exists(os.path.join(location, "plots")):
+                shutil.rmtree(os.path.join(location, "plots"))
+            if os.path.exists(os.path.join(location, "code")):
+                shutil.rmtree(os.path.join(location, "code"))
+            if os.path.exists(os.path.join(location, "plots.html")):
+                os.remove(os.path.join(location, "plots.html"))
+
+            if not (create or generate_code or all_plots or plot_list):
+                return
+
+            if not plot_list:
+                plot_list = [[self.lsec_drivers, self.lsec_layer_registry],
+                             [self.vsec_drivers, self.vsec_layer_registry],
+                             [self.hsec_drivers, self.hsec_layer_registry]]
+
+            for driver, registry in plot_list:
+                multiple_datasets = len(driver) > 1
+                for dataset in driver:
+                    plot_driver = driver[dataset]
+                    if dataset not in registry:
+                        continue
+                    for plot in registry[dataset]:
+                        plot_object = registry[dataset][plot]
+                        l_type = "Linear" if driver == self.lsec_drivers else \
+                            "Side" if driver == self.vsec_drivers else "Top"
+
+                        try:
+                            if not os.path.exists(os.path.join(location, "plots",
+                                                               f"{l_type}_{dataset if multiple_datasets else ''}"
+                                                               f"{plot_object.name}.png")):
+                                # Plot doesn't already exist, generate it
+                                file_type = next((field[0] for field in plot_object.required_datafields
```

Comment:
I dislike the use of next here. Many people are not that familiar with iterators. I would spend the extra line for better maintainability.

Reply:
I will abide.
I always found the use of next quite nice as an equivalent to what other languages offer out of the box for enumerables, particularly C#'s Enumerable.First(predicate), which is more performant for less code.
I don't often think in Python terms, so oddities like that arise occasionally.

Reply:
I concur that a "first" function would be very readable (or "car" for me, as I had a lisp/scheme phase). 
"next" is confusing IMO.
Even something like `([x for x in ...] + ["sfc"])[0]` would be better.

Writing code for other people/your older self is something one takes more into account when getting older ;-)